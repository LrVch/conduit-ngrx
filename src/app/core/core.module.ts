import { NgModule, Optional, SkipSelf } from '@angular/core';
import { CommonModule } from '@angular/common';
import { HTTP_INTERCEPTORS } from '@angular/common/http';
import { HttpTokenInterceptor } from './interceptors/http.token.interceptor';

import {
  ApiService,
  ArticlesService,
  AuthGuard,
  CommentsService,
  JwtService,
  ProfilesService,
  TagsService,
  UserService,
  CustomEventManager,
  ScrollService,
  DomUtilService
} from './services';
import { EventManager } from '@angular/platform-browser';

@NgModule({
  imports: [
    CommonModule
  ],
  providers: [
    { provide: HTTP_INTERCEPTORS, useClass: HttpTokenInterceptor, multi: true },
    ApiService,
    ArticlesService,
    AuthGuard,
    CommentsService,
    JwtService,
    ProfilesService,
    TagsService,
    UserService,
    ScrollService,
    DomUtilService,
    { provide: EventManager, useClass: CustomEventManager }
  ],
  declarations: []
})
export class CoreModule {
  constructor (@Optional() @SkipSelf() parentModule: CoreModule) {
    if (parentModule) {
      throw new Error(
        'CoreModule is already loaded. Import it in the AppModule only');
    }
  }
 }
