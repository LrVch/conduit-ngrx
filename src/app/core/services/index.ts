export * from './api.service';
export * from './articles.service';
export * from './auth-guard.service';
export * from './comments.service';
export * from './profiles.service';
export * from './tags.service';
export * from './user.service';
export * from './custom-event-manager';
export * from './main-loader.service';
export * from './scroll.service';
export * from './dom-util.service';
export * from './local-storage.service';
export * from './animations.service';
