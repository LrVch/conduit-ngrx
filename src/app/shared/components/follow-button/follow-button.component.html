<button mat-stroked-button
        [disabled]="isLoading"
        [color]="followed ? '': 'primary'"
        (click)="onFavoriteClick($event)">
  {{ followed ? ('conduit.followButton.unfollow' | translate) : ('conduit.followButton.follow' | translate) }}
  <ng-content></ng-content>
</button>