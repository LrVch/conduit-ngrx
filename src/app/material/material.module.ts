import { NgModule } from '@angular/core';
import {
  MatButtonModule,
  MatCheckboxModule,
  MatToolbarModule,
  MatCardModule,
  MatChipsModule,
  MatProgressBarModule,
  MatPaginatorModule,
  MatTabsModule,
  MatIconModule,
  MatInputModule,
  MatProgressSpinnerModule
} from '@angular/material';

const modules = [
  MatButtonModule,
  MatCheckboxModule,
  MatToolbarModule,
  MatCardModule,
  MatChipsModule,
  MatProgressBarModule,
  MatPaginatorModule,
  MatTabsModule,
  MatIconModule,
  MatInputModule,
  MatProgressSpinnerModule
];

@NgModule({
  imports: [...modules],
  exports: [...modules]
})
export class MaterialModule { }
