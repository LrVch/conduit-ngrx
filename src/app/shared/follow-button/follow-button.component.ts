import { Component, Input, Output, EventEmitter } from '@angular/core';

@Component({
  selector: 'app-follow-button',
  templateUrl: './follow-button.component.html',
  styleUrls: ['./follow-button.component.scss']
})
export class FollowButtonComponent {
  @Input('followed') followed = false;
  @Input('isLoading') isLoading = false;
  @Output() followedToggle = new EventEmitter();

  onFavoriteClick($event): void {
    $event.preventDefault();
    if (this.isLoading) {
      return;
    }

    this.followedToggle.emit();
  }
}
