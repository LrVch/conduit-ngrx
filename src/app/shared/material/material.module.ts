import { NgModule } from '@angular/core';
import {
  MatButtonModule,
  MatCheckboxModule,
  MatToolbarModule,
  MatCardModule,
  MatChipsModule,
  MatProgressBarModule,
  MatPaginatorModule,
  MatTabsModule,
  MatIconModule,
  MatInputModule,
  MatProgressSpinnerModule,
  MatDialogModule,
  MatSnackBarModule
} from '@angular/material';

const modules = [
  MatButtonModule,
  MatCheckboxModule,
  MatToolbarModule,
  MatCardModule,
  MatChipsModule,
  MatProgressBarModule,
  MatPaginatorModule,
  MatTabsModule,
  MatIconModule,
  MatInputModule,
  MatProgressSpinnerModule,
  MatDialogModule,
  MatSnackBarModule
];

@NgModule({
  imports: modules,
  exports: modules
})
export class MaterialModule { }
