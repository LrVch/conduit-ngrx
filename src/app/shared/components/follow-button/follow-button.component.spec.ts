import { async, ComponentFixture } from '@angular/core/testing';

import { ConfigureFn, configureTests } from '@app/lib/testing';

import { FollowButtonComponent } from './follow-button.component';
import { NO_ERRORS_SCHEMA, DebugElement } from '@angular/core';
import { By } from '@angular/platform-browser';
import { MaterialModule } from '@app/shared/material/material.module';

describe('FollowButtonComponent', () => {
    let component: FollowButtonComponent;
    let fixture: ComponentFixture<FollowButtonComponent>;
    let de: DebugElement;
    let el: HTMLElement;
    const event = {
        prevented: false,
        preventDefault() {
            this.prevented = true;
        }
    };

    beforeEach(
        async(() => {
            const configure: ConfigureFn = testBed => {
                testBed.configureTestingModule({
                    declarations: [FollowButtonComponent],
                    imports: [MaterialModule]
                });
            };

            configureTests(configure).then(testBed => {
                fixture = testBed.createComponent(FollowButtonComponent);
                component = fixture.componentInstance;
                de = fixture.debugElement;
                el = de.nativeElement;
                fixture.detectChanges();
            });
        })
    );

    it('should create', () => {
        expect(component).toBeTruthy();
    });

    it('should has "followed" @Input', () => {
        expect(component.followed).toBe(false);
    });

    it('should has "isLoading" @Input', () => {
        expect(component.isLoading).toBe(false);
    });

    it('should show "Unfollow" text by "followed" @Input equals to false', () => {
        component.followed = false;

        fixture.detectChanges();
        expect(fixture).toMatchSnapshot();
    });

    it('should show "Follow" text by "followed" @Input equals to true', () => {
        component.followed = true;

        fixture.detectChanges();
        expect(fixture).toMatchSnapshot();
    });

    it('should has a color that is equal to primary by "followed" @Input equals to false', () => {
        component.followed = false;

        fixture.detectChanges();
        expect(fixture).toMatchSnapshot();
    });

    it('shouldn\'t has a color that is equal to primary by "followed" @Input equals to false', () => {
        component.followed = false;

        fixture.detectChanges();
        expect(fixture).toMatchSnapshot();
    });

    it('should be disabled by "isLoading" @Input equals to true', () => {
        component.isLoading = true;

        fixture.detectChanges();
        expect(fixture).toMatchSnapshot();
    });

    it('shouldn\'t be disabled by "isLoading" @Input equals to false', () => {
        component.isLoading = false;

        fixture.detectChanges();
        expect(fixture).toMatchSnapshot();
    });

    it('should raises followedToggle event when clicked (onFavoriteClick)', () => {
        let raised = null;
        component.followedToggle.subscribe(_ => raised = true);

        fixture.detectChanges();

        const button = de.query(By.css('button'));
        button.triggerEventHandler('click', event);

        expect(raised).toBe(true);
    });

    it('shouldn\'t raises followedToggle event when clicked (onFavoriteClick) and "isLoading" @Input equals to true', () => {
        let wasEmited = false;

        component.isLoading = true;
        component.followedToggle.subscribe(_ => wasEmited = true);

        fixture.detectChanges();
        const button = de.query(By.css('button'));

        button.triggerEventHandler('click', event);

        expect(wasEmited).toBe(false);
    });

    it('should prevent defaul behaviour of clicked element when clicked (onFavoriteClick)', () => {
        component.followedToggle.subscribe();

        fixture.detectChanges();
        const button = de.query(By.css('button'));

        button.triggerEventHandler('click', event);

        expect(event.prevented).toBe(true);
    });
});

