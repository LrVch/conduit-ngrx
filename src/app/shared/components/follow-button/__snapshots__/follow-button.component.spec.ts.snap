// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FollowButtonComponent should be disabled by "isLoading" @Input equals to true 1`] = `
<app-follow-button
  followed="false"
  followedToggle={[Function EventEmitter]}
  isLoading={[Function Boolean]}
>
  <button
    class="mat-stroked-button mat-primary"
    disabled=""
    mat-stroked-button=""
    ng-reflect-color="primary"
    ng-reflect-disabled="true"
  >
    <span
      class="mat-button-wrapper"
    >
       conduit.followButton.follow 
    </span>
    <div
      class="mat-button-ripple mat-ripple"
      matripple=""
      ng-reflect-centered="false"
      ng-reflect-disabled="true"
      ng-reflect-trigger="[object HTMLButtonElement]"
    />
    <div
      class="mat-button-focus-overlay"
    />
  </button>
</app-follow-button>
`;

exports[`FollowButtonComponent should has a color that is equal to primary by "followed" @Input equals to false 1`] = `
<app-follow-button
  followed="false"
  followedToggle={[Function EventEmitter]}
  isLoading="false"
>
  <button
    class="mat-stroked-button mat-primary"
    mat-stroked-button=""
    ng-reflect-color="primary"
    ng-reflect-disabled="false"
  >
    <span
      class="mat-button-wrapper"
    >
       conduit.followButton.follow 
    </span>
    <div
      class="mat-button-ripple mat-ripple"
      matripple=""
      ng-reflect-centered="false"
      ng-reflect-disabled="false"
      ng-reflect-trigger="[object HTMLButtonElement]"
    />
    <div
      class="mat-button-focus-overlay"
    />
  </button>
</app-follow-button>
`;

exports[`FollowButtonComponent should show "Follow" text by "followed" @Input equals to true 1`] = `
<app-follow-button
  followed={[Function Boolean]}
  followedToggle={[Function EventEmitter]}
  isLoading="false"
>
  <button
    class="mat-stroked-button"
    mat-stroked-button=""
    ng-reflect-color=""
    ng-reflect-disabled="false"
  >
    <span
      class="mat-button-wrapper"
    >
       conduit.followButton.unfollow 
    </span>
    <div
      class="mat-button-ripple mat-ripple"
      matripple=""
      ng-reflect-centered="false"
      ng-reflect-disabled="false"
      ng-reflect-trigger="[object HTMLButtonElement]"
    />
    <div
      class="mat-button-focus-overlay"
    />
  </button>
</app-follow-button>
`;

exports[`FollowButtonComponent should show "Unfollow" text by "followed" @Input equals to false 1`] = `
<app-follow-button
  followed="false"
  followedToggle={[Function EventEmitter]}
  isLoading="false"
>
  <button
    class="mat-stroked-button mat-primary"
    mat-stroked-button=""
    ng-reflect-color="primary"
    ng-reflect-disabled="false"
  >
    <span
      class="mat-button-wrapper"
    >
       conduit.followButton.follow 
    </span>
    <div
      class="mat-button-ripple mat-ripple"
      matripple=""
      ng-reflect-centered="false"
      ng-reflect-disabled="false"
      ng-reflect-trigger="[object HTMLButtonElement]"
    />
    <div
      class="mat-button-focus-overlay"
    />
  </button>
</app-follow-button>
`;

exports[`FollowButtonComponent shouldn't be disabled by "isLoading" @Input equals to false 1`] = `
<app-follow-button
  followed="false"
  followedToggle={[Function EventEmitter]}
  isLoading="false"
>
  <button
    class="mat-stroked-button mat-primary"
    mat-stroked-button=""
    ng-reflect-color="primary"
    ng-reflect-disabled="false"
  >
    <span
      class="mat-button-wrapper"
    >
       conduit.followButton.follow 
    </span>
    <div
      class="mat-button-ripple mat-ripple"
      matripple=""
      ng-reflect-centered="false"
      ng-reflect-disabled="false"
      ng-reflect-trigger="[object HTMLButtonElement]"
    />
    <div
      class="mat-button-focus-overlay"
    />
  </button>
</app-follow-button>
`;

exports[`FollowButtonComponent shouldn't has a color that is equal to primary by "followed" @Input equals to false 1`] = `
<app-follow-button
  followed="false"
  followedToggle={[Function EventEmitter]}
  isLoading="false"
>
  <button
    class="mat-stroked-button mat-primary"
    mat-stroked-button=""
    ng-reflect-color="primary"
    ng-reflect-disabled="false"
  >
    <span
      class="mat-button-wrapper"
    >
       conduit.followButton.follow 
    </span>
    <div
      class="mat-button-ripple mat-ripple"
      matripple=""
      ng-reflect-centered="false"
      ng-reflect-disabled="false"
      ng-reflect-trigger="[object HTMLButtonElement]"
    />
    <div
      class="mat-button-focus-overlay"
    />
  </button>
</app-follow-button>
`;
